syntax = "proto3";

import "printer_system_state.proto";
import "pressure_control.proto";
import "print_control.proto";

message GetPrinterSystemStateRequest {

}

message ChangePrinterSystemStateRequest {
    PrinterSystemState state = 1;
}

message ChangeDropwatcherParametersRequest {
    int32 delay_nanos = 1;
    int32 flash_on_time_nanos = 2;
}

message CameraFrameRequest {
}

message ChangePressureControlParametersRequest {
    PressureControlParameters parameters = 1;
}

message ChangeNozzleDataRequest {
    repeated uint32 data = 1;
}

message ChangeEncoderPositionRequest {
    int32 position = 1;
}

message ChangeEncoderModeSettingsRequest {
    PrintControlEncoderModeSettings encoder_mode_settings = 1;
}

message PrinterRequest
{
    optional GetPrinterSystemStateRequest get_printer_system_state_request = 1;
    optional ChangePrinterSystemStateRequest change_printer_system_state_request = 2;
    optional ChangeDropwatcherParametersRequest change_dropwatcher_parameters_request = 3;
    optional CameraFrameRequest camera_frame_request = 4;
    optional ChangePressureControlParametersRequest change_pressure_control_parameter_request = 5;
    optional ChangeNozzleDataRequest change_nozzle_data_request = 6;
    optional ChangeEncoderPositionRequest change_encoder_position_request = 7;
    optional ChangeEncoderModeSettingsRequest change_encoder_mode_settings_request = 8;
}